library/src/main/java/org/xmtp/android/library/libxmtp/Message.kt

```diff
@@ -45,33 +44,34 @@ data class Message(val client: Client, private val libXMTPMessage: FfiMessage) {
             FfiDeliveryStatus.FAILED -> MessageDeliveryStatus.FAILED
         }
 
-    fun decode(): DecodedMessage {
-        try {
-            val decodedMessage = DecodedMessage(
-                id = id,
-                client = client,
-                topic = Topic.groupMessage(convoId).description,
-                encodedContent = EncodedContent.parseFrom(libXMTPMessage.content),
-                senderInboxId = senderInboxId,
-                sent = sentAt,
-                sentNs = sentAtNs,
-                deliveryStatus = deliveryStatus
-            )
-            if (decodedMessage.encodedContent.type == ContentTypeGroupUpdated && libXMTPMessage.kind != FfiConversationMessageKind.MEMBERSHIP_CHANGE) {
-                throw XMTPException("Error decoding group membership change")
-            }
-            return decodedMessage
-        } catch (e: Exception) {
-            throw XMTPException("Error decoding message", e)
+    val topic: String
+        get() = Topic.groupMessage(convoId).description
+
+    val encodedContent: Content.EncodedContent
+        get() = EncodedContent.parseFrom(libXMTPMessage.content)
+
+    fun <T> content(): T? =
+        encodedContent.decoded()
```

Comment:
I'm thinking the decoding should maybe happen in the constructor of the Message instead of in the content() function. That way, that way if you call content() on a message multiple times it doesnt repeat the decoding logic. wdyt?

Reply:
That makes sense!